### PR TITLE
fix: handle missing/null targeting keys in fractional evaluator

### LINKED ISSUE #1948: [BUG] missing targeting key handling differs across SDK implementations
## Parent Issue

Sub-issue of https://github.com/open-feature/flagd/issues/1904

## Summary

When fractional evaluation is used without an explicit bucketing property argument, bucketing should depend on the `targetingKey`. If the `targetingKey` is missing (or `null`) in the evaluation context, no bucket should be selected and the evaluator should return `null` so that the flag’s default variant (or code default, if no default variant exists) is used.

### Java

- targetingKey = null: explicit check, returns null
- targetingKey = "": treated as any other string (bucketing by flagkey+"")
- targetingKey missing entirely: is set to null -> same as targeting = null case

### Python

- targetingKey = nil: explicit check, returns null
- targetingKey = "": same as null
- targetingKey missing entirely: same as null

### JS

- targetingKey = null: explicit check, returns null
- targetingKey = "": same as null
- targetingKey missing entirely: same as null

### Go/flagd core

- targetingKey = nil: explicit check, returns null
- targetingKey = "": treated as a normal string (bucketing by flagkey+"")
- targetingKey missing entirely: same as null

### dotnet

- targetingKey = null: throws NullRef exception in `FlagdProperties.cs`
- targetingKey = "": treated as a normal string (bucketing by flagkey+"")
- targetingKey missing entirely: bucketing using flagKey only (`flagdProperties.FlagKey + flagdProperties.TargetingKey` -> treated as flagkey+"")

## EDIT: Suggested Fix

In all 3 cases (targeting key is null/nil, missing, or ""), we want to return null and fall back to the default variant.

### PR DESCRIPTION
**Pull request by utafrali**
Fixes #1948

The fractional evaluator wasn't handling missing or null targeting keys the same way other SDKs did. Added nil checks to return the default variant when the key is missing or null. Also added tests for the three cases: missing key, null key, and empty string.

### CODE REVIEW

**Comment by toddbaert** [comment]
Hey @utafrali  - can you do a `git commit --amend --signoff`? This is a legal requirement for CNCF projects. See the failing DCO action: https://github.com/open-feature/flagd/pull/1949/checks?check_run_id=71604066833

File: core/pkg/evaluator/fractional_test.go [hunk 1]
```diff
@@ -442,6 +442,67 @@ func TestFractionalEvaluation(t *testing.T) {
...
+				Targeting: []byte(`{
+							"fractional": [
+								["blue", 50],
+								["green", 50]
+							]
+						}`),
+			}},
+			flagKey: "headerColor",
+			context: map[string]any{
+				"targetingKey": "",
```
Comment (leakonvalinka) [inline core/pkg/evaluator/fractional_test.go:500]: Hi, this wasn't defined in the issue yet when you opened your PR, but the suggested fix is that `targetingKey = ""` should be treated the same as `targetingkey = nil` and a non-existent targeting key and therefore result in falling back to the default. :)

Comment (toddbaert) [reply core/pkg/evaluator/fractional_test.go:500]: +1, see the testbed test for this:
https://github.com/open-feature/flagd-testbed/blob/484ada4dd979f663214c024f3cb601141328a1b5/gherkin/targeting.feature#L376-L381

**Comment by utafrali** [comment]
Done, pushed the fix.

File: core/pkg/evaluator/fractional.go [hunk 2]
```diff
@@ -48,6 +48,10 @@ func (fe *Fractional) Evaluate(values, data any) any {
 		return nil
 	}
 
+	if feDistributions == nil {
```
Comment (leakonvalinka) [inline core/pkg/evaluator/fractional.go:51]: Can this ever be the case? 
Edit: Oh yes, I see, with your changes below.

File: core/pkg/evaluator/fractional.go [hunk 3]
```diff
@@ -78,11 +82,18 @@ if len(valuesArray) < 1 {
 			valuesArray = valuesArray[1:]
 		}
 
+		if dataMap[targetingKeyKey] == nil {
+			return "", nil, nil
+		}
 		targetingKey, ok := dataMap[targetingKeyKey].(string)
 		if !ok {
```
Comment (leakonvalinka) [inline core/pkg/evaluator/fractional.go:89]: Just a suggestion: You could also simply add a check here for an empty string.

**Review by leakonvalinka — APPROVED** [review]
I left a feedback comment, but just as a suggestion so not blocking of course!

**Comment by utafrali** [comment]
Done, pushed the fix.

**Comment by toddbaert** [comment]
> Hey @utafrali - can you do a `git commit --amend --signoff`? This is a legal requirement for CNCF projects. See the failing DCO action: https://github.com/open-feature/flagd/pull/1949/checks?check_run_id=71604066833

This still need to be done for both commits. @utafrali

**Comment by utafrali** [comment]
Done, pushed the fix.

**Review by beeme1mr — APPROVED** [review]
